packages/react-native/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultReactHost.kt

```diff
@@ -175,4 +175,12 @@ public object DefaultReactHost {
     }
     return reactNativeHost.toReactHost(context)
   }
+
+  /**
+    * Cleanup function for brownfield scenarios where you want to remove the references kept by
+    * reactHost after destroying the RN instance.
+    */
+  public fun invalidate() {
+      reactHost = null
+  }
```

Comment:
can you instead `override fun clear()`, call `super.clear()` and set the reactHost to null here?

Reply:
I am not sure I get it. `DefaultReactHost` does not inherit from `ReactNativeHost` so how can I override `clear` method in it? Or am I messing it up?

Reply:
@WoLewicki you're right, I misread the class title.

I'd suggest instead we do the following:
- Here make `public fun invalidate()` be `internal`
- Inside DefaultReactNativeHost add the following:

```kotlin
  override fun clear() {
    super.clear()
    DefaultReactHost.invalidate()
  }
```

Reply:
Done